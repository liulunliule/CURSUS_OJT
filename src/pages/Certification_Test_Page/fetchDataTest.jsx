import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { fetchQuestions } from "../../redux/features/certificationTestPageSlice";

export const useFetchQuestions = () => {
  const dispatch = useDispatch();
  const questions = useSelector(
    (state) => state.certificationTestPage.questions
  );
  const status = useSelector((state) => state.certificationTestPage.status);
  const error = useSelector((state) => state.certificationTestPage.error);

  useEffect(() => {
    if (status === "idle") {
      dispatch(fetchQuestions());
    }
  }, [status, dispatch]);

  return { questions, status, error };
};
