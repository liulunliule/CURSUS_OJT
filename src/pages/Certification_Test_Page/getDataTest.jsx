import { getQuestion } from "../../services/apiService";

export const hanldeGetQuestion = async () => {
    try {
        const response = await getQuestion();
        if (!response.ok) {
            throw new Error("Network response was not ok");
        }
        return await response.json();
    } catch (error) {
        console.error("Error fetching user reviews:", error);
        throw error;
    }
};
