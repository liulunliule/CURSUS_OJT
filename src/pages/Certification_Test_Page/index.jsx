import React, { useEffect, useState } from "react";
import "./index.scss";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAnglesLeft } from "@fortawesome/free-solid-svg-icons";
import { useDispatch, useSelector } from "react-redux";
import { fetchQuestions } from "../../redux/features/certificationTestPageSlice";

const Certification_Test_Page = () => {
  const dispatch = useDispatch();
  const { questions, status, error } = useSelector(
    (state) => state.certificationTestPage
  );
  useEffect(() => {
    dispatch(fetchQuestions());
  }, [dispatch]);
  return (
    <div className="Certification_test">
      <div className="toolbar_certification">
        <div className="container">
          <div className="row">
            <div className="col-lg-12">
              <div className="toolbar_title">
                <div className="titleleft">
                  <div className="toolbar_title_item">
                    <nav aria-label="breadcrumb">
                      <ol className="breadcrumb">
                        <li className="breadcrumb-item">
                          <Link to="/">Home</Link>
                        </li>
                        <li className="breadcrumb-item">
                          <Link to="/secondLayout/certification_center">
                            Certification Center
                          </Link>
                        </li>
                        <li
                          className="breadcrumb-item active"
                          aria-current="page"
                        >
                          WordPress Test View
                        </li>
                      </ol>
                    </nav>
                  </div>
                </div>
                <div className="titleright">
                  <Link
                    to="/secondLayout/certification_center"
                    className="blog_link"
                  >
                    <FontAwesomeIcon icon={faAnglesLeft} /> Back to
                    Certification Center
                  </Link>
                </div>
              </div>
              <div className="main_title">
                <h2>WordPress Test View</h2>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="container">
        <div className="row">
          <div className="col-lg-4 col-md-6">
            <div className="certi_form stickytime">
              <div className="test_timer_bg">
                <ul className="test_timer_left">
                  <li>
                    <div className="timer_time">
                      <h4>{questions.length}</h4>
                      <p>Questions</p>
                    </div>
                  </li>
                  <li>
                    <div className="timer_time">
                      <h4 id="timer">51:02</h4>
                      <p>Minutes</p>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div className="col-lg-7 col-md-6">
            <div className="certi_form">
              <div className="">
                {status === "loading" && <div>Loading...</div>}
                {status === "failed" && <div>{error}</div>}
                {status === "succeeded" &&
                  questions.length > 0 &&
                  questions.map((ques, index) => (
                    <div className="ques_item">
                      <div className="ques_title">
                        <span>Ques {index + 1} :-</span>
                        {ques.Question}
                      </div>
                      <div className="UI form">
                        <div className="gr_ed flds">
                          {ques.Kind_Of_Answer === "single choice" ? (
                            ques.Answer.map((option, idx) => (
                              <div className="fld fltr-radid" key={idx}>
                                <div className="UI radid ckbox">
                                  <input
                                    type="radio"
                                    name={`question${ques.id}`}
                                    tabIndex="0"
                                    className="hidden"
                                  />
                                  <label>{option}</label>
                                </div>
                              </div>
                            ))
                          ) : (
                            <div className="UI search focus mt-15">
                              <div className="UI form swdh30">
                                <div className="fld">
                                  <textarea
                                    rows="5"
                                    name="answer"
                                    id="id_answer"
                                    placeholder="Pls explain"
                                  ></textarea>
                                </div>
                              </div>
                            </div>
                          )}
                        </div>
                      </div>
                    </div>
                  ))}
              </div>
              <Link to="/secondLayout/Certification_test_result">
                <button className="test_submit_btn" type="submit">
                  Submit Test
                </button>
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Certification_Test_Page;
