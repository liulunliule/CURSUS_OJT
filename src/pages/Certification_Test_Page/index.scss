.Certification_test {
  font-family: 'Roboto', sans-serif;
  background-color: #f8f9fa;
  padding: 0;
  margin-top: 56px;
  
  .container {
    max-width: 1200px;
  }
}

.stickytime {
  position: sticky;
  top: 90px;
}

.test_timer_bg {
  background: #fff;
  width: 100%;
  float: left;
  padding: 0;
  border-radius: 3px;
  margin-bottom: 50px;
  border: 1px solid #efefef;
  transition: all .2s ease-in-out;
  
  .test_timer_left {
    width: 100%;
    float: left;
    display: flex;
    padding: 0;

    & li {
      width: 50%;
      float: left;
      display: inline-block;
      padding: 30px 20px;
      border-right: 1px solid #efefef;
    }

    .timer_time {
      float: left;
      width: 100%;
      text-align: center;

      & h4 {
        font-size: 24px;
        font-weight: 600;
        color: #333;
        margin-bottom: 15px;
        text-align: center;
        line-height: 26px;
        height: 100px;
        width: 100px;
        display: inline-block;
        border: 2px solid #333;
        border-radius: 100%;
        padding: 35px 0;
        background: #fff;

        &:first-child {
          margin-top: auto;
        }
      }

      & p {
        font-size: 14px;
        font-weight: 400;
        color: #333;
        margin-top: 0;
        text-align: center;
        line-height: 26px;

        &:last-child {
          margin-bottom: 0;
        }
      }
    }
  }
}

.ques {
  &_item {
    margin-bottom: 35px;
  }

  &_title {
    font-size: 16px;
    font-weight: 500;
    color: #333;
    text-align: left;
    line-height: 28px;
    margin-top: 0;
    margin-bottom: 24px;

    & .ques_title span {
      font-weight: 600;
      margin-right: 10px;
    }
  }
}

.UI.form {
  font-size: 1rem;
  position: relative;
  max-width: 100%;
  
  .gr_ed.flds {
    display: block;
    margin: 0 0 1em;
    
    .fld {
      display: block;
      margin: .5em 0;
      padding: 0;
    }

    &:last-child {
      margin-bottom: 0;
    }
  }

  & textarea {
    background: #fff;
    border: 1px solid #e5e5e5 !important;
    outline: 0;
    color: rgba(0, 0, 0, 0.87);
    border-radius: 3px !important;
    box-shadow: inset 0px 1px 4px rgba(0, 0, 0, 0.02) !important;
    vertical-align: top;
    width: 100%;
    padding-top: 0.785714em;
    padding-right: 1em;
    padding-bottom: 0.785714em;
    padding-left: 1em;
    box-sizing: border-box;

    &:focus {
      border: 1px solid #333 !important;
      box-shadow: none;
    }
  }
}

.test_submit_btn {
  height: 40px;
  padding: 0 20px;
  border: 0;
  margin-top: 30px;
  background: #333;
  color: #fff;
  font-size: 14px;
  font-weight: 500;
  border-radius: 3px;
  text-transform: uppercase;
  letter-spacing: .5px;
  cursor: pointer;
  outline: none;
  transition: all .3s ease-in-out;

  &:hover {
    background: #444;
  }
}

.breadcrumb {
  background: none;
  padding: 0;
  margin-bottom: 0;

  .breadcrumb-item + .breadcrumb-item::before {
    content: '>';
    color: #6c757d;
    padding: 0 5px;
  }

  .breadcrumb-item a {
    color: #6c757d;
    text-decoration: none;

    &:hover {
      color: #333;
    }
  }
}

.main_title {
  text-align: center;
  margin: 20px 0;

  h2 {
    font-size: 30px;
    font-weight: 600;
    color: #333;
  }
}

.toolbar_certification {
  background: #f1f1f1;
  padding: 15px 0;
  border-bottom: 1px solid #e5e5e5;

  .container {
    display: flex;
    align-items: center;
    justify-content: space-between;

    .toolbar_title {
      display: flex;
      align-items: center;
      justify-content: space-between;
      width: 100%;

      .titleleft {
        display: flex;
        align-items: center;

        .toolbar_title_item {
          margin-left: 20px;
        }
      }

      .titleright {
        .blog_link {
          color: #007bff;
          text-decoration: none;
          display: flex;
          align-items: center;

          svg {
            margin-right: 5px;
          }

          &:hover {
            color: #0056b3;
          }
        }
      }
    }

    .main_title {
      text-align: center;
      margin: 20px 0;

      h2 {
        font-size: 30px;
        font-weight: 600;
        color: #333;
      }
    }
  }
}