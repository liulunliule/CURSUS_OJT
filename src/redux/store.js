import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
