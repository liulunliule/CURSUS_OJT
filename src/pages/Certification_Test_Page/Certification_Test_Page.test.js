// Certification_Test_Page.test.jsx
import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Certification_Test_Page from '.';

test('test Certification Test Page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Certification_Test_Page/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
