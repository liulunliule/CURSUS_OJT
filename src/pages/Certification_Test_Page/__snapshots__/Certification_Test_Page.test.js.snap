// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Certification Test Page 1`] = `
<div
  className="Certification_test"
>
  <div
    className="toolbar_certification"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <div
            className="toolbar_title"
          >
            <div
              className="titleleft"
            >
              <div
                className="toolbar_title_item"
              >
                <nav
                  aria-label="breadcrumb"
                >
                  <ol
                    className="breadcrumb"
                  >
                    <li
                      className="breadcrumb-item"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </li>
                    <li
                      className="breadcrumb-item"
                    >
                      <a
                        href="/secondLayout/certification_center"
                        onClick={[Function]}
                      >
                        Certification Center
                      </a>
                    </li>
                    <li
                      aria-current="page"
                      className="breadcrumb-item active"
                    >
                      WordPress Test View
                    </li>
                  </ol>
                </nav>
              </div>
            </div>
            <div
              className="titleright"
            >
              <a
                className="blog_link"
                href="/secondLayout/certification_center"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-angles-left "
                  data-icon="angles-left"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160zm352-160l-160 160c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L301.3 256 438.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                 Back to Certification Center
              </a>
            </div>
          </div>
          <div
            className="main_title"
          >
            <h2>
              WordPress Test View
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-4 col-md-6"
      >
        <div
          className="certi_form stickytime"
        >
          <div
            className="test_timer_bg"
          >
            <ul
              className="test_timer_left"
            >
              <li>
                <div
                  className="timer_time"
                >
                  <h4>
                    0
                  </h4>
                  <p>
                    Questions
                  </p>
                </div>
              </li>
              <li>
                <div
                  className="timer_time"
                >
                  <h4
                    id="timer"
                  >
                    59
                    :
                    59
                  </h4>
                  <p>
                    Minutes
                  </p>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="col-lg-7 col-md-6"
      >
        <div
          className="certi_form"
        >
          <div
            className=""
          />
          <a
            href="/secondLayout/Certification_test_result"
            onClick={[Function]}
          >
            <button
              className="test_submit_btn"
              type="submit"
            >
              Submit Test
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
